script.js

```diff
@@ -1,3 +1,27 @@
+figureManager = new function () {
+    this.update = function () {
+        $('.figures button').addClass('btn btn-danger').attr('type', 'button').attr('onclick', 'figureManager.remove(this);');
+        var i = 0;
+        $('.figures tr').each(function () {
+            $('input', this).eq(0).addClass('form-control').attr('type', 'text').attr('name', 'problem[figures][' + i +  '][path]');
+            $('input', this).eq(1).addClass('form-control').attr('type', 'text').attr('name', 'problem[figures][' + i +  '][caption]');
+            i++;
+        });
+    };
+
+    this.add = function () {
+        $('.figures').append('<tr><td><input></td><td><input></td><td><button>X</button></td></tr>');
+        this.update();
+    };
```

Comment:
existuje $.prototype.clone() čo je podľa mňa krajšie riešenie, tým oklonuješ elementy a nemusiš riešiť atributy a ako sa renderujú

Reply:
Když tam nic nebude, tak to nebudu moct odkud klonovat.

Reply:
Vždy tam predsa môže byť jeden prázdny, ktorý sa nedá odobrať. resp. odobrenie riešiť kontrolou "prázdnosti" na serveri.